### PR TITLE
feat: use operationName as the function name of the compiled output

### LINKED ISSUE #19: Name new Function to identify queries
Functions created in the executor are not named which makes hard to debug problematic queries.

- [ ] Add persisted query as functionName if possible
- [ ] Otherwise use the operationName

### PR DESCRIPTION
**Pull request by boopathi**
+ fix #19

### CODE REVIEW

**Comment by ruiaraujo** [comment]
:+1:

File: src/execution.ts [hunk 1]
```diff
@@ -212,87 +220,106 @@ export function createBoundQuery(
...
+   * we create an intermediate object with a method named as the
+   * intended function name. This is because Function.prototype.name
+   * is not writeable.
+   *
+   * http://www.ecma-international.org/ecma-262/6.0/#sec-method-definitions-runtime-semantics-propertydefinitionevaluation
+   *
+   * section: 14.3.9.3 - calls SetFunctionName
+   */
+  /* tslint:enable */
+  const ret = {
```
Comment (addityasingh) [inline src/execution.ts]: nit: I would use a more meaningful name such as `intermediateObject`. `ret` doesn't tells much about the intended purpose

Comment (boopathi) [reply src/execution.ts]: `ret` is a common variable name to indicate value used for **`ret`urn** . The comment above explains other things of using the object.

**Comment by boopathi** [comment]
👍

**Pull request merged**